### DISCUSSION TITLE
Upgrade di homeassistant core in venv

### DISCUSSION THREAD

**Discussion by vinny74man** [indomus/forum#1874] [HUB: Home Assistant]
Ciao a tutti come da titolo dovrei fare l'upgrade di HA, attualmente sono alla versione  2022.3.5 e dovrei installare HACS che però mi richiede la versione 2022.8 eseguendo l'upgrade posso compromettere la configurazione attuale? Ho come storage mysql con tutti i dati energia che non posso assolutamente perdere, quindi prima di fare l'upgrade voglio essere sicuro di non fare danni, per quanto possibile.

Grazie

**Comment by snakuzzo** [comment]
Questa è la dimostrazione che tocca essere sempre al passo e che conviene sempre aggiornare in breve tempo. Nessuno qui può dirti con certezza che aggiorni e funziona tutto. La tua configurazione la sai solo tu.
Devi metterti lì, controllare i changelog di tutte le versioni, fino all'ultima che vuoi installare e adegui la configurazione.
Se non ricordo male, mi sa che devi anche installare python alla 3.10 e quindi ricreare il venv (altro motivo per cui questo tipo di installazione è sconsigliata per chi non è pratico e conviene utilizzare Docker).
In linea di massima il db non dovresti perderlo, però ricorda che tra le varie versioni sono state fatte delle ottimizzazioni e cambia la struttura...fatti sempre un backup

**Reply by vinny74man** [reply]
Io dovrei solo installare il custom component bar-card e non trovo il modo di installarlo senza hacs se c'è un modo non ho motivo di aggiornare, se qualcuno può aiutarmi a fare un'installazione manuale di questo custom-component.

[bar-card](https://github.com/custom-cards/bar-card)

Grazie

**Reply by snakuzzo** [reply]
È una semplice custom card, non è un custom component. Si installa come qualsiasi custom card.
Prendi l'ultimo file js rilasciato sotto Releases e lo piazzi nella dir (o in una subdir) sotto `/config/www` e poi vai su HA sotto `SETTINGS > DASHBOARD > RESOURCES` e lo referenzi indicando path e tipologia (JavaScript).
Ricorda che la dir www viene referenziata con local. Se ad esempio piazzi il js sotto www lì indicherai ...

```
/local/bar-card.js
```

**Reply by vinny74man** [reply]
Il punto è che nel repository di git-hub non riesco a trovare il file bar-card.js, potresti aiutarmi a capire dove cercarlo
Grazie

**Reply by vinny74man** [reply]
Scusa l'ho trovato non sono molto pratico di git e non avevo visto la cartella release.

Grazie 1000

**Reply by snakuzzo** [reply]
Sì ma, piccolo particolare... git non è github
